packages/workspace/src/node/default-workspace-server.ts

```diff
@@ -154,51 +155,37 @@ export class DefaultWorkspaceServer implements WorkspaceServer {
     /**
      * Reads the most recently used workspace root from the user's home directory.
      */
-    private async readMostRecentWorkspaceRootFromUserHome(): Promise<WorkspaceData | undefined> {
-        const data = await this.readJsonFromFile(this.getUserStoragePath());
-        if (data && WorkspaceData.is(data)) {
+    private async readRecentWorkspacePathsFromUserHome(): Promise<RecentWorkspacePathsData | undefined> {
+        const filePath = this.getUserStoragePath();
+        const data = await this.readJsonFromFile(filePath);
+        if (data && RecentWorkspacePathsData.is(data)) {
             return data;
         }
+        const message = `Unable to retrieve recent workspaces from the file: '${filePath}'. Please check if the file is corrupted.`;
```

Comment:
I saw this message when starting Theia initially.

Reply:
ah, it happens because the format of the workspace data has been changed from 
`{folders:[f1, f2, ..., fn]}` 
to 
`{folders: {path: f1}, {path: f2}, ..., {path: f3}}`. 

Do we need to have something like a "fall-back" strategy (i.e., if the new format doesn't work, use the old format to parse the data)?

Reply:
Ok, understood. I think it's fine to go without compatibility still, given the relatively low number of installments